presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueryFramework.java

```diff
@@ -172,6 +176,23 @@ protected void assertUpdate(Session session, @Language("SQL") String sql, long c
         QueryAssertions.assertUpdate(queryRunner, session, sql, OptionalLong.of(count));
     }
 
+    protected void assertQueryFailsEventually(@Language("SQL") String sql, @Language("RegExp") String expectedMessageRegExp, Duration timeout)
```

Comment:
Ideally, the logic for this method should be in `QueryAssertions` instead (with an optional `assertQueryFailsEventually()` method in `AbstractTestQueryFramework` to make things easier).